src/views/Wizard/views/DataplaneUniversal.vue

```diff
@@ -662,32 +611,35 @@ export default {
         univDataplaneNetworkProtocol
       } = this.validate
 
-      // topology step field validation
-      if (step === 1) {
-        if (
-          univDataplaneServiceName &&
-            univDataplaneId
-        ) {
-          this.nextDisabled = false
+      if (!mesh.length) {
+        return true
+      }
+
+      if (this.$route.query.step === '1') {
+        if (univDataplaneServiceName && univDataplaneId) {
+          return false
         } else {
-          this.nextDisabled = true
+          return true
         }
       }
 
-      // network step field validation
-      if (step === 2) {
+      if (this.$route.query.step === '2') {
         if (
           univDataplaneNetworkAddress &&
             univDataplaneNetworkServicePort &&
             univDataplaneNetworkDPPort &&
             univDataplaneNetworkProtocol
         ) {
-          this.nextDisabled = false
+          return false
         } else {
-          this.nextDisabled = true
+          return true
         }
```

Comment:
This can be simplified and if you don"t want to return with this complicated condition, we can get read of at least the `else` statement. Wdyt?